store/sqlstore/thread_store.go

```diff
@@ -483,35 +483,55 @@ func (s *SqlThreadStore) GetThreadForUser(teamId string, threadMembership *model
 			sq.Eq{"Posts.DeleteAt": 0},
 		})
 
-	fetchConditions := sq.And{
-		sq.Or{
-			sq.Eq{"Threads.ThreadTeamId": teamId},
-			// Include DMs & GMs
-			sq.Eq{"Threads.ThreadTeamId": ""},
-		},
```

Comment:
Apologies, but I think I'm missing something in this change overall: why not just remove this `sq.Or` and leave the existing queries otherwise intact?

Reply:
The difference is filtering first and joining later or joining first and filtering later. On most cases query optimizer does the "right thing" but not always. As a rule I tend to right optimized queries(not rely on optimizer that much). In this concrete case I benchmark tested and both solutions perform pretty much the same. So I'm reverting to the original code.